HabitTracker.GreyKlouds/HabitTracker/Program.cs

```diff
@@ -0,0 +1,248 @@
+    static string connectionString = @"Data Source=habitTracker.db";
+    static void Main(string[] args)
+    {
+        // establish the connection to our database
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            // open the connection and call the createCommand
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            // create a table with a sql query
+            tableCmd.CommandText =
+            @"CREATE TABLE IF NOT EXISTS DrinkingWater
+            (
+                Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                Date TEXT,
+                Quantity INTEGER
+            )";
+            // call the execute not query
+            tableCmd.ExecuteNonQuery();
+            // close teh connection
+            connection.Close();
+
+        }
+        GetUserinput();
+    }
+    static void GetUserinput()
+    {
+        Console.Clear();
+        bool closeApp = false;
+        while (closeApp == false)
+        {
+            Console.WriteLine("\t\tMAIN MENU");
+            Console.WriteLine("-----------------------------------------");
+            Console.WriteLine("\nWhat would you like to do?");
+            Console.WriteLine("1 -> Insert A Record");
+            Console.WriteLine("2 -> Update A Record");
+            Console.WriteLine("3 -> Delete A Record.");
+            Console.WriteLine("4 -> View All Records.");
+            Console.WriteLine("5 -> Close Application");
+            Console.WriteLine("-----------------------------------------");
+
+            string userInput = Console.ReadLine();
+
+            switch (userInput)
+            {
+                case "1":
+                    Insert();
+                    break;
+                case "2":
+                    Update();
+                    break;
+                case "3":
+                    Delete();
+                    break;
+                case "4":
+                    GetAllRecords();
+                    break;
+                case "5":
+                    Console.WriteLine("Goodbye!");
+                    closeApp = true;
+                    Environment.Exit(0);
+                    break;
+                default:
+                    Console.WriteLine("Invalid command. Press any key to start over");
+                    Console.ReadKey();
+                    Console.Clear();
+                    break;
+            }
+        }
+    }
+    private static void Insert()
+    {
+        string date = GetDateInput();
+
+        int quantity = GetNumberInput("\n\nPlease insert number of glasses or mesurment of your choice (Whole numbers only)\n\n");
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+            $"INSERT INTO DrinkingWater(date,quantity) VALUES('{date}',{quantity})";

```

Comment:
🟠 **Parameterised SQL Queries**

💡 I know the project allows for this. But I really feel that you should always be parameterising your SQL queries. It takes little refactor, and the benefits you get outweigh all.

🔧
❌ 
```
command.CommandText = $"INSERT INTO Table(Column) VALUES ('{value}');";
```
✔️ 
```
command.CommandText = $"INSERT INTO Table(Column) VALUES (@Value);";
command.Parameters.AddWithValue("@Value", value);
```